Diagnostics/AVTester/Test-ExchAVExclusions.ps1

```diff
@@ -313,6 +314,34 @@ $currentDiff = $initialDiff
 $firstExecution = $true
 $SuspiciousProcessList = New-Object Collections.Generic.List[string]
 $SuspiciousW3wpProcessList = New-Object Collections.Generic.List[string]
+$SuspiciousAMSIinW3wpProcessList = New-Object Collections.Generic.List[string]
+
+# Get AMSI Dlls registered
+# Define the AMSI providers registry path
+$registryPath = "HKLM:\SOFTWARE\Microsoft\AMSI\Providers"
+
+# Get all subKeys in the specified registry path
+$subKeys = Get-ChildItem -Path $registryPath -ErrorAction SilentlyContinue
+
+if ($subKeys) {
+    # Regular expression to match the subKey names
+    $regex = "[0-9A-Fa-f\-]{36}"
+
+    # Filter the subKeys that match the regular expression and get only their names
+    $matchingSubKeys = $subKeys | Where-Object { $_.Name -match $regex } | ForEach-Object { $_.PSChildName }
+
+    if ($matchingSubKeys) {
+        $AMSIDll = $null
+        $AMSIDll = $matchingSubKeys | ForEach-Object { (Get-Item "HKLM:\SOFTWARE\Classes\ClSid\$_\InprocServer32" -ErrorAction SilentlyContinue).GetValue("").trim('"') }
+        if ($null -eq $AMSIDll) {
+            Write-Host 'No AMSI Dlls was found for $subKeys, possible AMSI misconfiguration"' -ForegroundColor Red
```

Comment:
`$subKeys` appears to be a list, therefore, when converting to a string this isn't going to be useful if it is a list.